### PR TITLE
`KernelSummary::validations`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR adds `KernelSummary::validations`, a vector of `(boolean_expr, string_reason)` pairs. This field will serve as a unified framework for validations, and will also be used for validating TMA. The previous `KernelSummary::splits_to_validate` is removed and the corresponding validations are moved to this new framework.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/kernel.h [hunk 1]
```diff
@@ -92,8 +92,8 @@ struct KernelSummary {
   //! Only used for debugging.
   std::vector<const kir::Allocate*> dynamic_lmem_allocations;
 
-  //! ceilDiv extents that must be divisible
-  std::vector<std::pair<const Val*, const Val*>> splits_to_validate;
+  //! Validations needed and information about them
```
Comment (naoyam) [inline csrc/kernel.h:95]:
```suggestion
  //! Validations needed and information about them. For example, a pair of "extent mod split_factor == 0" and an error message for divisibility check for vectorization.
```

**Review by naoyam — APPROVED** [review]
LGTM. Just a suggestion to add more info.

**Pull request merged**